source/code-snippets/usage-examples/find.js

```diff
@@ -0,0 +1,41 @@
+
+const { MongoClient } = require('mongodb');
+
+
+const client = new MongoClient(
+   'mongodb+srv://admin:admin@mflix-2sp0m.mongodb.net/test?retryWrites=true&w=majority',
+);
+
+async function run() {
+   try {
+      await client.connect();
+
+      const database = client.db('sample_mflix');
+      const collection = database.collection('movies');
+
+      // create a query document for a range match for a movie less than 15 minutes long
+      const query = { runtime: { $lt : 15 } };
+
+      // initialize an empty options object
+      options = {}
+
+      // sort our returned documents by ascending title
+      const sort = { title: 1 };
+
+      // returned documents should only contain title and the imdb object, no _id
+      const projection = { title : 1, _id: 0, imdb: 1 }
+      options.sort = sort;
+      options.projection = projection;
+
+      // pass our query and constructed options
+      const cursor = collection.find(query, options);
+
+      // iterate with await so the session doesn't end while printing
+      for await (const movie of cursor) {
+         console.dir(movie);
+      }
```

Comment:
What if the query does not match any documents or throws an error?

Reply:
Why do we need to iterate with `await`? (Because the Cursor is just a pointer to documents in the result set, so we need to actually go and fetch each one)

Reply:
- if the query does not match any documents, it doesn't fail but it also iterates over zero items, so nothing really happens. Not sure if there's a better way (maybe check length, if 0 print "no matches found?")
- if the query throws an error, it's caught in the catch() at the bottom of the file
- i agree with your last point, but it's a little long to include in a comment. I think I'd prefer to explain that part of the Cursor object in the text and then keep it simple (this plus a little reference to that?) in the comments on the code itself. Thoughts? @nlarew